diff -ruN wineditline-WinEditLine-2.206/src/CMakeLists.txt wineditline-WinEditLine-2.206-mod/src/CMakeLists.txt
--- wineditline-WinEditLine-2.206/src/CMakeLists.txt	2022-05-04 00:08:43
+++ wineditline-WinEditLine-2.206-mod/src/CMakeLists.txt	2023-05-30 21:54:40
@@ -13,16 +13,15 @@
 add_executable(edit_test libedit_test.c)
 target_link_libraries(edit_test edit)
 add_executable(edit_test_dll libedit_test_dll.c)
-install (TARGETS edit edit_test edit_test_dll
-  DESTINATION "${CMAKE_SOURCE_DIR}/bin${LIB_SUFFIX}")
-install (TARGETS edit_a
-  DESTINATION "${CMAKE_SOURCE_DIR}/lib${LIB_SUFFIX}")
-install (FILES editline/readline.h
-  DESTINATION "${CMAKE_SOURCE_DIR}/include/editline")
-string(TOUPPER "${CMAKE_BUILD_TYPE}" uppercase_CMAKE_BUILD_TYPE)
-if (MSVC AND uppercase_CMAKE_BUILD_TYPE MATCHES "DEBUG")
-  install (FILES ${CMAKE_CURRENT_BINARY_DIR}/Debug/edit_a.pdb
-    DESTINATION "${CMAKE_SOURCE_DIR}/lib${LIB_SUFFIX}")
-  install (FILES ${CMAKE_CURRENT_BINARY_DIR}/Debug/edit.pdb
-    DESTINATION "${CMAKE_SOURCE_DIR}/bin${LIB_SUFFIX}")
-endif()
+
+install(TARGETS edit
+    RUNTIME DESTINATION "bin"
+    ARCHIVE DESTINATION "lib"
+    LIBRARY DESTINATION "lib")
+
+install(TARGETS edit_a
+    RUNTIME DESTINATION "bin"
+    ARCHIVE DESTINATION "lib"
+    LIBRARY DESTINATION "lib")
+
+install(FILES editline/readline.h DESTINATION "include/editline")
