diff -ruN IP2Location-C-Library-8.5.1/Makefile.am IP2Location-C-Library-8.5.1-mod/Makefile.am
--- IP2Location-C-Library-8.5.1/Makefile.am	2022-07-14 16:40:16
+++ IP2Location-C-Library-8.5.1-mod/Makefile.am	2023-05-10 20:56:34
@@ -1,7 +1,7 @@
 bin_PROGRAMS=ip2location
 
 ip2location_SOURCES=ip2location.c libIP2Location/IP2Location.c
-ip2location_LDADD=-lrt
+ip2location_LDADD=
 ip2location_CFLAGS=-IlibIP2Location -Wall
 
 dist_man_MANS=ip2location.1
